dcicutils/scripts/update_portal_object.py

```diff
@@ -0,0 +1,430 @@
+# ------------------------------------------------------------------------------------------------------
+# Command-line utility to update (post, patch, upsert) portal objects for SMaHT/CGAP/Fourfront.
+# ------------------------------------------------------------------------------------------------------
+# Example commands:
+# update-portal-object --post file_format.json
+# update-portal-object --upsert directory-with-schema-named-dot-json-files
+# update-portal-object --patch file-not-named-for-schema-name.json --schema UnalignedReads
+# --------------------------------------------------------------------------------------------------
+
+import argparse
+from functools import lru_cache
+import glob
+import io
+import json
+import os
+import sys
+from typing import Callable, List, Optional, Tuple, Union
+from dcicutils.command_utils import yes_or_no
+from dcicutils.common import ORCHESTRATED_APPS, APP_SMAHT
+from dcicutils.ff_utils import delete_metadata, purge_metadata
+from dcicutils.misc_utils import get_error_message, PRINT
+from dcicutils.portal_utils import Portal as PortalFromUtils
+
+
+class Portal(PortalFromUtils):
+
+    def delete_metadata(self, object_id: str) -> Optional[dict]:
+        if isinstance(object_id, str) and object_id and self.key:
+            return delete_metadata(obj_id=object_id, key=self.key)
+        return None
+
+    def purge_metadata(self, object_id: str) -> Optional[dict]:
+        if isinstance(object_id, str) and object_id and self.key:
+            return purge_metadata(obj_id=object_id, key=self.key)
+        return None
+
+
+_DEFAULT_APP = "smaht"
+_SMAHT_ENV_ENVIRON_NAME = "SMAHT_ENV"
+
+# Schema properties to ignore (by default) for the view schema usage.
+_SCHEMAS_IGNORE_PROPERTIES = [
+    "date_created",
+    "last_modified",
+    "principals_allowed",
+    "submitted_by",
+    "schema_version"
+]
+
+_SCHEMA_ORDER = [  # See: smaht-portal/src/encoded/project/loadxl.py
```

Comment:
What do you use this for? I would recommend not doing it this way, instead add an API in smaht-portal that returns this and have this info come in via a call to the server